File: src/450-删除二叉搜索树中的节点/index1.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} key
 * @return {TreeNode}
 */
 var deleteNode = function(root, key) {
  if(!root) return null

  if(root.val > key) {
      root.left = deleteNode(root.left, key)
  } else if(root.val < key) {
      root.right = deleteNode(root.right, key)
  } else {
      if(!root.left) return root.right
      if(!root.right) return root.left

      var node = root.left
      while(node.right) node = node.right

      // 找到前驱节点
      var temp = node
      // 进行覆盖
      root.val = temp.val
      root.left = deleteNode(root.left, temp.val)
  }

  return root
};

File: src/145-二叉树的后序遍历/index1.py

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    """
    二叉树的后序遍历，使用栈的方式
    """

    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        if root is None:
            return res
        stack = [root]
        while len(stack):
            root = stack.pop()
            res = [root.val] + res
            if root.left:
                stack.append(root.left)
            if root.right:
                stack.append(root.right)

        return res
